### PR TITLE
Added DamageTypes to Repairing

### PR DESCRIPTION
**Pull request by Mailaender**
This sounds a bit counter-intuitive, but it is a pre-requisite for https://github.com/OpenRA/ra2/issues/148.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/RepairsUnits.cs [hunk 1]
```diff
@@ -23,6 +24,9 @@ public class RepairsUnitsInfo : PausableConditionalTraitInfo
 		[Desc("Time (in ticks) between two repair steps.")]
 		public readonly int Interval = 24;
 
+		[Desc("Damage type used for the repair.")]
+		public readonly BitSet<DamageType> RepairDamageType = default(BitSet<DamageType>);
```
Comment (MustaphaTR) [inline OpenRA.Mods.Common/Traits/RepairsUnits.cs]: RepairDamageTypes would be better as this can take mutiple values.

Comment (Mailaender) [reply OpenRA.Mods.Common/Traits/RepairsUnits.cs]: Good idea.

File: OpenRA.Mods.Common/Traits/Buildings/RepairableBuilding.cs [hunk 2]
```diff
@@ -28,6 +29,9 @@ public class RepairableBuildingInfo : ConditionalTraitInfo, Requires<IHealthInfo
 		[Desc("The maximum amount of HP to repair each step.")]
 		public readonly int RepairStep = 7;
 
+		[Desc("Damage type used for the repair.")]
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/Buildings/RepairableBuilding.cs]: Nit: `types`

File: OpenRA.Mods.Common/Traits/RepairsUnits.cs [hunk 3]
```diff
@@ -23,6 +24,9 @@ public class RepairsUnitsInfo : PausableConditionalTraitInfo
 		[Desc("Time (in ticks) between two repair steps.")]
 		public readonly int Interval = 24;
 
+		[Desc("Damage type used for the repair.")]
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/RepairsUnits.cs]: Same here.

**Comment by Mailaender** [comment]
Pluralized.

**Review by abcdefg30 — APPROVED** [review]
Ok.

**Comment by atlimit8** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/05a7aeca32c3237a3c95d87bd5fea74a8720f7e4)

**Pull request merged**